js/index.js

```diff
@@ -1,135 +1,104 @@
-					removeBtn = document.createElement('button');
-		
-			// Add props to elems
-			link.innerHTML = item.name;
-			link.className = 'js-link-item'
-			removeBtn.className = 'secondary-btn kill-item-btn js-remove';
-			removeBtn.innerHTML = 'x';
-
-			// Build the list item
-			linkItem.appendChild(link);
-			linkItem.appendChild(removeBtn);
-			menuList.appendChild(linkItem);
-		});
-	}
-
-	// Remove list from DOM
-	const removeList = function removeList() {
-		let listContainer = document.querySelectorAll('.content-container')[1];
-		listContainer.remove();
-	}
-
-	// Remove individual list item
-	const removeListItem = function removeListItem(e) {
-		let listItem = e.target.closest('li');		
-		listItem.remove();
-	}
-	
-	const refreshItem = function refreshItem(arr) {
-		// compare arrays and return unrepeated obj to replace current
-	}
-
-	// const showDetails = function showDetails(e) {
-	// 	let listItem = this.closest('li');
-	// }
-	// 	const myMeals = Meals.filter(function(whom){ return whom.owner === 'Brian'}).map(function(whom){ return whom.name; });
-
+(function () {
+    'use strict';
+
+    const GENERATE_BTN = document.querySelector('#GenerateMealList');
+    const URL = 'http://www.thebrianpye.com/Recipes/meals.json';
+    const BTN_TEXT = [
+        `Make it again`,
+        `Nope, try again`,
+        `Keep on trying`,
+        `I Don't like this` //Template literals preserve whitespace and quotation marks.
+    ];
+
+    let count = 0;
+    GENERATE_BTN.addEventListener('click', (ev) => main(ev));
+
+
+    function main(ev) {
+        const listContainer = document.querySelectorAll('.content-container')[1]; //I tried setting it to query an id, but I kept getting undefined errors for some reason. Leaving it alone for now.
+        const days = 7;
+        const newMessage = BTN_TEXT[Math.floor(Math.random() * BTN_TEXT.length)];
+
+        if (listContainer !== undefined) {
+            removeList();
+        }
+
+        GENERATE_BTN.textContent = newMessage;
+        count++;
+        if (count % 10 === 0) {
+            GENERATE_BTN.textContent = `Are you effing serious?`;
+        }
+        //Replaced the callbacks (which I don't really understand) and XHR with fetch(). 
+        fetch(URL).then(response => response.json()).then(response => createList(response));
+
+        function createList(response) {
+            selectRandomMeals(days, response, createListHTML);
+            const list = document.querySelector('.js-list');
+            const removeBtnArr = document.querySelectorAll('.js-remove');
+            const linkItemArr = document.querySelectorAll('.js-link-item');
+            removeBtnArr.forEach((item) => {
+                const button = item.closest('button');
+                button.addEventListener('click', removeListItem);
+            });
+        }
+    }
+
+
+
+    function selectRandomMeals(days, arr, callback) {
+        const randomlyOrderedArray = arr.sort(() => {
```

Comment:
`arr.sort(() => {}`
So arrow functions are simply anonymous functions that replace the word `function` with `=>`? Or is there more to it than that?